src/clj/rems/db/attachments.clj

```diff
@@ -1,112 +1,156 @@
 (ns rems.db.attachments
-  (:require [clojure.test :refer [deftest is]]
-            [clojure.tools.logging :as log]
-            [rems.common.attachment-util :as attachment-util]
-            [rems.common.util :refer [fix-filename]]
-            [rems.config :refer [env]]
+  (:require [clj-time.core :as time]
+            [rems.cache :as cache]
+            [rems.common.util :refer [getx index-by]]
             [rems.db.core :as db]
-            [rems.multipart :refer [scan-for-malware]]
-            [rems.util :refer [file-to-bytes]])
-  (:import [rems PayloadTooLargeException UnsupportedMediaTypeException InvalidRequestException]))
+            [schema.core :as s]
+            [schema.coerce :as coerce]))
 
-(defn check-size
-  [file]
-  (when (= :too-large (:error file)) ; set already by the multipart upload wrapper
-    (throw (PayloadTooLargeException. (str "File is too large")))))
+(s/defschema AttachmentDb
+  {:attachment/id s/Int
+   :attachment/user s/Str
+   :attachment/filename s/Str
+   :attachment/type s/Str
+   :application/id s/Int})
 
-(defn check-allowed-attachment
-  [filename]
-  (when-not (attachment-util/allowed-extension? filename)
-    (throw (UnsupportedMediaTypeException. (str "Unsupported extension: " filename)))))
+(def ^:private coerce-AttachmentDb
+  (coerce/coercer! AttachmentDb coerce/string-coercion-matcher))
+
+(defn- parse-attachment-raw [x]
+  (coerce-AttachmentDb {:application/id (:appid x)
+                        :attachment/id (:id x)
+                        :attachment/user (:userid x)
+                        :attachment/filename (:filename x)
+                        :attachment/type (:type x)}))
+
+(def attachment-cache
```

Comment:
cache naming could reflect the fact it contains metadata? e.g. `attachment-metadata-cache`

Reply:
could also add comment about metadata cache _not_ containing actual attachment data.